expression/builtin_convert_charset.go

```diff
@@ -65,6 +66,10 @@ func (c *tidbToBinaryFunctionClass) getFunction(ctx sessionctx.Context, args []E
 			return nil, err
 		}
 		bf.tp = args[0].GetType().Clone()
+		// adjust enum type
+		if bf.tp.Tp == mysql.TypeEnum && !mysql.HasEnumSetAsIntFlag(bf.tp.Flag) {
+			bf.tp = types.NewFieldType(mysql.TypeString)
```

Comment:
1. Please check mysql.TypeSet
2. Don't need to check HasEnumSetAsIntFlag, because argTp is ETString
3. bf.tp.Tp = mysql.TypeString is ok...